src/hotspot/cpu/x86/frame_x86.cpp

```diff
@@ -720,20 +712,14 @@ frame::frame(void* sp, void* fp, void* pc) {
 
 #endif
 
-void JavaFrameAnchor::make_walkable(JavaThread* thread) {
+void JavaFrameAnchor::make_walkable() {
   // last frame set?
   if (last_Java_sp() == NULL) return;
   // already walkable?
   if (walkable()) return;
-  vmassert(Thread::current() == (Thread*)thread, "not current thread");
+  vmassert(Thread::current() == thread, "not current thread");
```

Comment:
Oops! Need to delete this.

Reply:
... as well as the assertion that follows.